src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import duke.command.Command;
+import duke.ui.Ui;
+
+/**
+ * Main class for the duke application.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isExit;
+
+    public Duke() {
+        this("data/duke.txt");
+    }
+
+    /**
+     * Constructs a Duke object and initialises the Storage, Ui and TaskList.
+     *
+     * @param filePath The file path where the tasks are stored.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        isExit = false;
```

Comment:
if false is the default value for isExit, you can define it in the variable declaration itself